apps/tenant-management-webapp-e2e/src/integration/calendar/calendar.step.ts

```diff
@@ -0,0 +1,247 @@
+        .calendarTableBody()
+        .find('tr')
+        .then((rows) => {
+          rows.toArray().forEach((rowElement) => {
+            let counter = 0;
+            // cy.log(rowElement.cells[0].innerHTML); // Print out the name cell innerHTML for debug purpose
+            if (rowElement.cells[0].innerHTML.includes(name)) {
+              counter = counter + 1;
+            }
+            // cy.log(rowElement.cells[2].innerHTML); // Print out the description cell innerHTML for debug purpose
+            if (rowElement.cells[2].innerHTML.includes(desc)) {
+              counter = counter + 1;
+            }
+            // cy.log(rowElement.cells[3].innerHTML); // Print out the role cell innerHTML for debug purpose
+            roles.forEach((runningRole) => {
+              if (rowElement.cells[3].innerHTML.includes(runningRole.trim())) {
+                counter = counter + 1;
+              }
+            });
+            Cypress.log({
+              name: 'Number of matched items for row# ' + rowElement.rowIndex + ': ',
+              message: String(String(counter)),
+            });
+            if (counter == targetedNumber) {
+              rowNumber = rowElement.rowIndex;
+            }
+          });
+          Cypress.log({
+            name: 'Row number for the found script: ',
+            message: String(rowNumber),
+          });
+          resolve(rowNumber);
+        });
+    } catch (error) {
+      reject(error);
+    }
+  });
+}
+
+When(
+  'the user clicks {string} button for the calendar of {string}, {string}, {string}',
+  function (button, name, desc, role) {
+    findCalendar(name, desc, role).then((rowNumber) => {
+      expect(rowNumber).to.be.greaterThan(
+        0,
+        'Script of ' + name + ', ' + desc + ', ' + role + ' has row #' + rowNumber
+      );
+      cy.wait(1000); // Wait for buttons to show up
+      switch (button.toLowerCase()) {
+        case 'edit':
+          calendarObj.calendarEditButton(rowNumber).shadow().find('button').click({ force: true });
+          break;
+        case 'delete':
+          calendarObj.calendarDeleteButton(rowNumber).shadow().find('button').click({ force: true });
+          break;
+        default:
+          expect(button).to.be.oneOf(['edit', 'delete']);
+      }
+    });
+  }
+);
+
+Then('the user views Edit calendar modal', function () {
+  calendarObj.editCalendarModal().should('exist');
+});
+
+When('the user enters {string} as description in Edit calendar modal', function (description) {
+  calendarObj.editCalendarModalDescriptionField().shadow().find('.goa-textarea').clear().type(description, { force: true });
+});
+
+When('the user select {string} as role in Edit calendar modal', function (role) {
+  // calendarObj.editCalendarModalDescriptionField().shadow().find('.goa-textarea').clear().type(description, { force: true });
+  const roles = role.split(',');
+    for (let i = 0; i < roles.length; i++) {
+      if (roles[i].includes(':')) {
```

Comment:
Missing the steps to check if the checkbox is already checked or not. The assumption of the current checkbox being in unchecked status adds a limitation of calling this step. What we usually do in other services is to clear all role checkboxes and select the ones (complete list) passed in as role parameter.